### PR TITLE
include nireq during streams calculation

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *Include nireq during streams calculation. If nireq is smaller than nstreams, will reduce nstreams to nireq and keep using all CPU resource*

### Tickets:
 - *106343*

### CODE REVIEW

File: src/plugins/intel_cpu/src/cpu_streams_calculation.hpp [hunk 1]
```diff
@@ -15,10 +15,13 @@ namespace ov {
 namespace intel_cpu {
 /**
  * @brief      Generate streams information table according to processors type table
- * @param[in]  input_streams is target streams set by user via NUM_STREAMS or hints.
+ * @param[in]  input_streams is target streams set by user via ov::num_streams or hints.
```
Comment (wenjiew) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.hpp:18]: input_streams is the targeted number of sreams set by user...

File: src/plugins/intel_cpu/src/cpu_streams_calculation.hpp [hunk 2]
```diff
@@ -15,10 +15,13 @@ namespace ov {
...
  *               - input "0" mean function generate the optimal number of streams
- *               - LATENCY hint equals 1 stream.
- * @param[in]  input_threads is max threads set by user via INFERNECE_NUM_THREADS.
+ *               - LATENCY hint equals 1 stream on one socket platform.
+ * @param[in]  input_threads is max threads set by user via ov::inference_num_threads.
```
Comment (wenjiew) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.hpp:21]: input_threads is the max number of threads set by user...

File: src/plugins/intel_cpu/src/cpu_streams_calculation.hpp (hunk 1 above)
Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.hpp]: updated

File: src/plugins/intel_cpu/src/cpu_streams_calculation.hpp (hunk 2 above)
Comment (wangleis) [reply src/plugins/intel_cpu/src/cpu_streams_calculation.hpp]: updated

File: src/plugins/intel_cpu/src/cpu_streams_calculation.cpp [hunk 3]
```diff
@@ -107,21 +107,29 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
...
+                    while (n_streams < n_threads_per_stream) {
+                        if (1 == n_threads_per_stream) {
+                            break;
+                        } else {
+                            n_threads_per_stream = static_cast<int>((n_threads_per_stream * 2 - 1) / 2);
+                            n_threads_per_stream =
+                                static_cast<int>(proc_type_table[0][MAIN_CORE_PROC] /
+                                                 ((proc_type_table[0][MAIN_CORE_PROC] + n_threads_per_stream - 1) /
+                                                  n_threads_per_stream));
+                            n_streams = static_cast<int>(n_threads / n_threads_per_stream);
```
Comment (sunxiaoxia2022) [inline src/plugins/intel_cpu/src/cpu_streams_calculation.cpp:124]: I think it doesn't need "static_cast<int>".

**Review by sunxiaoxia2022 — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]
In general LGTM.
The follow-up that have to be done is documentation. We need to clarly specify CPU behavior related to streams calculation inside user documentation: https://docs.openvino.ai/latest/openvino_docs_OV_UG_supported_plugins_CPU.html#doxid-openvino-docs-o-v-u-g-supported-plugins-c-p-u. 
Please create corresponding ticket in bounds of CVS-106563

**Comment by wangleis** [comment]
@dmitry-gorokhov Sure, will follow up documentation in separated PR

**Pull request merged**